#!/usr/bin/env python

PACKAGE = 'gazelle_sim'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# Numeric values using sliders
# General syntax is:
#   gen.add('var_name', type, 0, 'Display Name', default, low, high)

# Robot control parmaeter
gen.add('robot_name',       str_t,    0, 'Robot Name',  '')
gen.add('camera_z',       double_t,   0, 'Camera Height',  0.5,   0,      2)
gen.add('camera_phi_deg', double_t,   0, 'Camera Rot Deg',  60,   0,     90)
gen.add('camera_f',       double_t,   0, 'Focal Len',      700,   0,   5000)

exit(gen.generate(PACKAGE, 'gazelle_sim', 'GazelleRobotParamDynCfg'))
